pkg/gameservers/controller.go

```diff
@@ -334,8 +334,24 @@ func (ext *Extensions) creationMutationHandlerPod(review admissionv1.AdmissionRe
 
 	ext.baseLogger.WithField("pod.Name", pod.Name).Debug("creationMutationHandlerPod")
 
-	// TODO: We need to deal with case of multiple and mixed type ports before enabling the feature gate.
-	pod.Spec.Containers[1].Ports[0].ContainerPort = pod.Spec.Containers[1].Ports[0].HostPort
+	annotation, ok := pod.ObjectMeta.Annotations[agonesv1.PassthroughPortAssignmentAnnotation]
+	if !ok {
+		ext.baseLogger.WithField("pod.Name", pod.Name).Debug("creationMutationHandlerPod, " + agonesv1.PassthroughPortAssignmentAnnotation + ": " + pod.ObjectMeta.Annotations[agonesv1.PassthroughPortAssignmentAnnotation])
+	}
+
+	passthroughPortAssignmentMap := make(map[string][]int)
+	if err := json.Unmarshal([]byte(annotation), &passthroughPortAssignmentMap); err != nil {
+		return review, errors.Wrapf(err, "could not unmarshal annotation %q (value %q)", passthroughPortAssignmentMap, annotation)
+	}
+
+	for _, p := range pod.Spec.Containers {
+		if containerPassthrough, ok := passthroughPortAssignmentMap[p.Name]; ok {
+			for i := range containerPassthrough {
+				i := containerPassthrough[i]
```

Comment:
Two things:
1) line 349/350 are a weird pairing because you're ranging over the indices, then immediately shadowing with the value. This can be shortened to just: `for _, i := range containerPassthrough`

That said, though, I would shy away from `i` here and probably make it slightly more descriptive, like `portIdx`.

2) Fun fact, `passthroughPortAssignmentMap[p.Name]` will default to `nil`, and a `nil` slice is an empty slice, so you can skip 348.

Basically I think line 348-350 can be replaced with:
```
for _, portIdx := range passthroughPortAssignmentMap[p.Name] {
}
```

Reply:
Okok sg, I changed the for loop variable name and replaed lines 348-350.